File: matlab/rk_rgb_tool.m
%%
clear; close all;

%%
path = '/media/manu/kingstop/itx-3588j/Linux_SDK/rk3588/external/linux-rga/samples/sample_file/in0w1280-h720-rgba8888.bin';

w = 1280;
h = 720;

c = 4;
s = w * h * c;

fileID = fopen(path);
A = fread(fileID, s, 'uint8')';
fclose(fileID);

R = uint8(reshape(A(1:c:end), w, h))';
G = uint8(reshape(A(2:c:end), w, h))';
B = uint8(reshape(A(3:c:end), w, h))';
A = uint8(reshape(A(4:c:end), w, h))';

RGB = cat(3, R, G, B);

imshow(R);

%%

File: tools/rk_rgb_tool.py
import numpy as np
import cv2

if __name__ == "__main__":
    path = \
        '/media/manu/kingstop/itx-3588j/Linux_SDK/rk3588/external/linux-rga/samples/sample_file/in0w1280-h720-rgba8888.bin'

    w, h, c = 1280, 720, 4

    data = np.fromfile(path, np.uint8)

    rgba = data.reshape(h, w, c)
    bgr = cv2.cvtColor(rgba, cv2.COLOR_RGBA2BGR)

    cv2.imshow('img', bgr)
    cv2.waitKey(0)
